src/ws-api/ws-api.ts

```diff
@@ -0,0 +1,84 @@
+import {
+    AnyFunction,
+    AnyFunctionWithThis,
+    OnMessageCb,
+    WebsocketClient,
+    WsProviderConnector
+} from '../connector/ws'
+import {ActiveOrdersWebSocketApi} from './active-websocket-orders-api'
+import {RpcWebsocketApi} from './rpc-websocket-api'
+import {WsApiConfigWithNetwork} from './types'
+import {castUrl} from './url'
+
+export class WebSocketApi {
+    public readonly rpc: RpcWebsocketApi
+
+    public readonly order: ActiveOrdersWebSocketApi
+
+    public readonly provider: WsProviderConnector
+
+    constructor(
+        configOrProvider: WsApiConfigWithNetwork | WsProviderConnector
+    ) {
+        if (configOrProvider instanceof WebsocketClient) {
```

Comment:
This part will not works with different provider implementation
Try to use something like this:
```
function instanceOfWsApiConfigWithNetwork(val: WsApiConfigWithNetwork | WsProviderConnector): val is WsApiConfigWithNetwork {
    return ('url' in val && 'network' in val)
}
```